src/contentType.ts

```diff
@@ -0,0 +1,35 @@
+import { fileTypeFromBuffer } from 'file-type'
+import mime from 'mime-types'
+
+interface testInput {
+  bytes: Uint8Array
+  path: string
+}
+
+type testOutput = Promise<string | undefined>
+
+export const DEFAULT_MIME_TYPE = 'text/html'
+
+const tests: Array<(input: testInput) => testOutput> = [
+  // testing file-type from buffer
+  async ({ bytes }): testOutput => (await fileTypeFromBuffer(bytes))?.mime,
+  // testing file-type from path
+  // ts-expect-error
+  async ({ path }): testOutput => mime.lookup(path) || undefined,
+  // svg
+  async ({ bytes }): testOutput => new TextDecoder().decode(bytes.slice(0, 4)) === '<svg' ? 'image/svg+xml' : undefined,
+  // default
+  async (): Promise<string> => DEFAULT_MIME_TYPE
+]
+
+const overrides: Record<string, string> = {
+  'video/quicktime': 'video/mp4'
+}
+
+export async function parseContentType (input: testInput): Promise<string> {
+  let type = (await Promise.all(tests.map(async test => test(input)))).filter(Boolean)[0] as string
```

Comment:
I feel like this is much harder to read than it needs to be. Also, we're *always* calling each of the type check fns (map), and then looping over all of the successful ones (filter), instead of returning quickly on the first success.